src/Geno/C++/Components/Workspace.cpp

```diff
@@ -122,10 +121,27 @@ bool Workspace::Deserialize( void )
 
 //////////////////////////////////////////////////////////////////////////
 
+void Workspace::Rename( const std::string& rName )
+{
+	const std::filesystem::path OldPath = ( m_Location / m_Name ).replace_extension( EXTENSION );
+
+	if( std::filesystem::exists( OldPath ) )
+	{
+		const std::filesystem::path NewPath = ( m_Location / rName ).replace_extension( EXTENSION );
+		std::filesystem::rename( OldPath, NewPath );
+	}
+
+	m_Name = std::move( rName );
```

Comment:
Surprised this compiles, but you can't move a const reference. Consider making it a simple `std::string Name`